aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -2709,6 +3271,542 @@ module aptos_framework::delegation_pool {
+        delegator1: &signer,
+        delegator2: &signer,
+        voter1: &signer,
+        voter2: &signer,
+    ) acquires DelegationPoolOwnership, DelegationPool, GovernanceRecords {
+        initialize_for_test_no_reward(aptos_framework);
+        aptos_governance::initialize_for_test(
+            aptos_framework,
+            (10 * ONE_APT as u128),
+            100 * ONE_APT,
+            1000,
+        );
+        aptos_governance::initialize_partial_voting(aptos_framework);
+
+        initialize_test_validator(validator, 100 * ONE_APT, true, false);
+
+        let validator_address = signer::address_of(validator);
+        let pool_address = get_owned_pool_address(validator_address);
+        // Delegation pool is created before partial governance voting feature flag is enabled. So this delegation
+        // pool's voter is its owner.
+        assert!(stake::get_delegated_voter(pool_address) == validator_address, 1);
+        assert!(!partial_governance_voting_enabled(pool_address), 1);
+
+        let delegator1_address = signer::address_of(delegator1);
+        account::create_account_for_test(delegator1_address);
+        let delegator2_address = signer::address_of(delegator2);
+        account::create_account_for_test(delegator2_address);
+        let voter1_address = signer::address_of(voter1);
+        account::create_account_for_test(voter1_address);
+        let voter2_address = signer::address_of(voter2);
+        account::create_account_for_test(voter2_address);
+
+        stake::mint(delegator1, 110 * ONE_APT);
+        add_stake(delegator1, pool_address, 10 * ONE_APT);
+        stake::mint(delegator2, 110 * ONE_APT);
+        add_stake(delegator2, pool_address, 90 * ONE_APT);
+
+        // Enable partial governance voting feature flag.
+        features::change_feature_flags(aptos_framework, vector[features::get_partial_governance_voting()], vector[]);
+        // Voter doens't change until enabling partial governance voting on this delegation pool.
+        assert!(stake::get_delegated_voter(pool_address) == validator_address, 1);
+        // Enable partial governance voting on this delegation pool.
+        enable_partial_governance_voting(pool_address);
+        assert!(stake::get_delegated_voter(pool_address) == pool_address, 1);
+        assert!(partial_governance_voting_enabled(pool_address), 1);
+
+        // By default, the voter of a delegator is itself.
+        assert!(calculate_and_update_voter_total_voting_power(pool_address, voter1_address) == 0, 1);
+        assert!(calculate_and_update_voter_total_voting_power(pool_address, voter2_address) == 0, 1);
+        assert!(calculate_and_update_voter_total_voting_power(pool_address, delegator1_address) == 10 * ONE_APT, 1);
+        assert!(calculate_and_update_voter_total_voting_power(pool_address, delegator2_address) == 90 * ONE_APT, 1);
+
+        // Delegator1 delegates its voting power to voter1. Delegator2 delegates its voting power to voter2.
```

Comment:
Do we need two delegators for this test or can we keep it simple and just have one?